src/soroban/assembled_transaction.ts

```diff
@@ -49,36 +54,78 @@ class Err<E extends ErrorMessage> implements Result<never, E> {
   isErr() { return true }
 }
 
-export type MethodOptions = {
+export type MethodOptions = AcceptsWalletOrAccount & {
   /**
    * The fee to pay for the transaction. Default: BASE_FEE
    */
   fee?: number;
+  /**
+   * The maximum amount of time to wait for the transaction to complete. Default: {@link DEFAULT_TIMEOUT}
+   */
+  timeoutInSeconds?: number;
```

Comment:
To move the discussion from here over:

> Do you think this is important enough to bring to people's attention that we should uglify every method call in this way, or do you think there's a good-enough default?

Honestly? Yes. I'm open to opinions from users but from a security standpoint, this should be an explicit choice every time. Also from a consistency standpoint, you have to call `setTimebounds`/`setTimeout` on your `TransactionBuilder` no matter what and intentionally opt-in to `TimeoutInfinite` if you decide to make that (bad) choice.

Reply:
I want to make sure we're on the same page—this change removes the `TimeoutInfinite` default. People _would_ need to opt into that. The default is now ten seconds. 

If you already understood that, sorry for over-explaining. If that's new information, does it change your opinion?

Reply:
Ahh thanks for clarifying - I guess I missed that. I don't think there should be **any** default at all.

Reply:
That's a huge change from current TS Bindings behavior and I'd like to hold off on it.

Reply:
works for me :+1: then a non-inf default is better